app/models/piece.rb

```diff
@@ -73,33 +68,29 @@ def is_obstructed_horizontally(position_x, position_y)
     end
   end
 
-  # 5. See if there is a vertical obstruction
   def is_obstructed_diagonally(position_x, position_y)
     x_current = piece.position_x
     y_current = piece.position_y
     x_destination = position_x
     y_destination = position_y
 
     if x_current < x_destination && y_current < y_destination # up-right diagonal
-      while x_current < x_destination && y_current < y_destination do |x, y|
-        return true if is_occupied?(x += 1)(y += 1) == true
-        end
+      while x_current < x_destination && y_current < y_destination do
+        return true if is_occupied?((x += 1),(y += 1))
```

Comment:
no need for `x += 1`, it can just be `x+1`.. we are passing the value of `x + 1` to the method `is_occupied?`.. += actually changes the x in the existing method.. does not have any consequences because its is in the return statement but the side effects are there..